app/assets/javascripts/sumofus/backbone/braintree_hosted_fields.js

```diff
@@ -34,27 +39,31 @@ const HostedFieldsMethods = {
             "font-size": "16px",
           },
         },
-        onFieldEvent: (event) => {
-          if (event.type === "fieldStateChange"){
-            if (event.isPotentiallyValid) {
-              this.clearError(event.target.fieldKey);
-            } else {
-              this.showError(event.target.fieldKey, I18n.t('errors.probably_invalid'));
-            }
-            if (event.target.fieldKey == 'number') {
-              if (event.isEmpty) {
-                this.$('#hosted-fields__paypal').removeClass('paypal--grayed-out');
-              } else {
-                this.$('#hosted-fields__paypal').addClass('paypal--grayed-out');
-              }
-            }
-            this.showCardType(event.card);
-          }
-        },
+        onFieldEvent: this.fieldUpdate(),
       },
     };
   },
 
+  fieldUpdate() {
+    return (event) => {
```

Comment:
I see this a lot, where you return a function (for closure so that `this` is what you expect it to be), but can't you use `bind`?:

```javascript
fieldUpdate() {
  ...
 onFieldEvent: this.fieldUpdate.bind(this)
  ...
},

fieldUpdate(event) {
  this.foo;
}
```